Clarify linker options in the guide are for GNU BFD

thomasnyman opened the pull request:
Fixes #923

thesamesam commented on docs/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.md:
```diff
@@ -157,6 +157,8 @@ The recommendations in Table 1 and Table 2 are primarily applicable to compiling
 - supported both by the GCC and Clang / LLVM toolchains.
 - cross-platform and supported on (at least) Intel and AMD 64-bit x86 architectures as well as the 64-bit version of the ARM architecture (AArch64).
 
+Linker options covered by this guide are valid for the GNU BFD linked (LD) provided by Binutils (unless otherwise noted).
```
```suggestion
Linker options covered by this guide are valid for the GNU BFD linker (LD) provided by Binutils (unless otherwise noted).
```

06kellyjac reviewed (approved):
Potential alternative wording. But current proposed wording also LGTM.

thesamesam commented on docs/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.md:
```diff
@@ -157,6 +157,8 @@ The recommendations in Table 1 and Table 2 are primarily applicable to compiling
 - supported both by the GCC and Clang / LLVM toolchains.
 - cross-platform and supported on (at least) Intel and AMD 64-bit x86 architectures as well as the 64-bit version of the ARM architecture (AArch64).
 
+Linker options covered by this guide are valid for the GNU BFD linker (LD) provided by Binutils (unless otherwise noted).
```
"GLD" isn't a commonly used term, and it's also ambiguous (could refer to the gold linker).

thomasnyman commented:
Incorporated @06kellyjac's suggestion (with @thesamesam's clarification) in 1e39dca although I opted to flip the order so that we refer to ld primarily (as the sentence starts with mentioning _linker_ options). Also added a trailing sentence to be more more specific that the options may not work with gold or lld.
